Dockerfile

```diff
@@ -1234,8 +1234,10 @@ ARG ADDITIONAL_PYTHON_DEPS=""
 # https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
 # * authlib, gcloud_aio_auth, adal are needed to generate constraints for PyPI packages and can be removed after we release
 #   new google, azure providers
+# * cloudpickle==2.0.0 is required by snowfalke-snowpark-python, which conflicts with new versions of apache beam, so
+#   we pin apache-beam also to the latest version which uses cloudpickle==2.0.0.
 # !!! MAKE SURE YOU SYNCHRONIZE THE LIST BETWEEN: Dockerfile, Dockerfile.ci, find_newer_dependencies.py
-ARG EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS="dill<0.3.3 pyarrow>=6.0.0 protobuf<4.21.0 authlib>=1.0.0 gcloud_aio_auth>=4.0.0 adal>=1.2.7"
+ARG EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS="dill<0.3.3 pyarrow>=6.0.0 protobuf<4.21.0 authlib>=1.0.0 gcloud_aio_auth>=4.0.0 adal>=1.2.7 cloudpickle==2.0.0 apache-beam==2.39.0"
```

Comment:
i think this is not good to `pin` the requirements to specific versions. if we really want to limit them, we should rather say <= or < here. 

Why do we have "apache-beam==2.39.0" and "cloudpickle==2.0.0" - seems a bit odd choice and also it will heavily limit our capabilities. I believe the ARM image failing is caused by this one and also we are waiting for the new version of apache-beam to be able to release Python 3.11. I think limiting those **just** to have snowpark decorator is pretty wrong. Is there any reason snowpark cannot work with higher (also future) versions of those? 

I think - to be perfectly honest, if that would be the consequence of accepting the snowpark, then possibly snowflake should release their own snowpark decorator as add-on to the existing provider? Limiting us to potentially fixed and vulnerable versions of dependencies is not something we should treat lightly.

Reply:
@mik-laj where does teh dependency for apache beam come from?

Here is wha I see the snowpark dependency list is?

asn1crypto==1.5.1
certifi==2022.12.7
cffi==1.15.1
charset-normalizer==2.1.1
cloudpickle==2.0.0
cryptography==38.0.4
filelock==3.9.0
idna==3.4
oscrypto==1.3.0
pycparser==2.21
pycryptodomex==3.16.0
PyJWT==2.6.0
pyOpenSSL==22.1.0
pytz==2022.7
requests==2.28.1
snowflake-connector-python==2.9.0
snowflake-snowpark-python==1.0.0
typing_extensions==4.4.0
urllib3==1.26.14

@mpgreg cc

Reply:
@mpgreg it looks like cloudpickle is also blocking here ... not apache-beam specifically